src/views/darkroom.c

```diff
@@ -812,7 +812,11 @@ static void _fire_darkroom_image_loaded_event(const bool clean, const int32_t im
 static void _dev_change_image(dt_develop_t *dev, const int32_t imgid)
 {
   // stop crazy users from sleeping on key-repeat spacebar:
-  if(dev->image_loading) return;
+  dt_pthread_mutex_lock(&dev->pipe_mutex);
+  const gboolean is_loading = dev->image_loading;
```

Comment:
Mutexes are the same as atomics - *every* access to a variable needs to be protected by the same lock or you can have data races.

If dt were written entirely in C++, we could just declare the variable atomic and have the compiler ensure that all accesses use atomic instructions, but for plain C the programmer has to explicitly change all of the accesses.

Reply:
> Mutexes are the same as atomics - _every_ access to a variable needs to be protected by the same lock or you can have data races.

Yes, and that's the case. During the run of the pipe in the proper section in `dt_dev_process_image_job()` the lock is taken. So this patch ensure that we do not try to reload an image until the previous one has been loaded.

Reply:
But this is maybe too conservative anyway and I'm currently testing without this part anyway.

Reply:
New version without this pushed now. This part was indeed not needed after all.

Reply:
And thanks for the review!